### PR TITLE
Feature/validate proposal

### PR DESCRIPTION
**Pull request by patrykbuzowiczsabre**

### CODE REVIEW

File: src/components/viewProposal/ValidationResultModal.tsx [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+export default class ValidationResultModal extends React.PureComponent<ValidationResultModalProps> {
+  render() {
+    const {onClosed, validationData, formatMessage} = this.props;
+
+    const isModalOpen = validationData.isDisplayed;
+    const hasLoaded = !validationData.isValidating;
+
+    return (
+      <Modal isOpen={isModalOpen}>
+        {hasLoaded &&
```
Comment (wojciech-kumon) [inline src/components/viewProposal/ValidationResultModal.tsx]: It should be more readable with only one `hasLoaded` check for spinner and actual modal (ternary operator)

File: src/containers/ViewTripProposalPage.tsx [hunk 2]
```diff
@@ -29,7 +31,7 @@ export class ViewTripProposalPage extends React.PureComponent<ViewTripProposalPa
   }
 
   render() {
-    const {isFetching, error, proposalData, proposalId, history, intl: {formatMessage}, actions} = this.props;
+    const {isFetching, error, proposalView, proposalId, history, intl: {formatMessage}, actions, validationData} = this.props;
```
Comment (wojciech-kumon) [inline src/containers/ViewTripProposalPage.tsx]: Try to shorten/split this line please

File: src/containers/ViewTripProposalPage.tsx [hunk 3]
```diff
@@ -48,15 +50,19 @@ export class ViewTripProposalPage extends React.PureComponent<ViewTripProposalPa
...
       result = (
         <>
           <ProposalHeader formatMessage={formatMessage}
                           proposalId={proposalId}
                           validate={actions.validateProposal} />
-          <TripProposalPreview iframeContent={proposalData.htmlContent} />
+          <TripProposalPreview iframeContent={proposalView.htmlContent} />
+
+          <ValidationResultModal onClosed={actions.dismissValidationResult}
+                                 validationData={validationData as ValidationData}
```
Comment (wojciech-kumon) [inline src/containers/ViewTripProposalPage.tsx]: remove this cast (change `error: any;` to `error?: any;` in `ValidationData` interface)

File: src/__tests__/reducers/viewProposalReducer.test.ts [hunk 4]
```diff
@@ -90,3 +101,67 @@ describe('viewProposal reducer', () => {
...
+  };
+
+  const errorData = {
+    message: 'fatality'
+  };
+
+  it('should set isValidating on validation start', () => {
+    const nextState = viewProposalReducer(initialState, validateProposalStart());
+
+    // TODO assert isDisplayed in below tests
```
Comment (wojciech-kumon) [inline src/__tests__/reducers/viewProposalReducer.test.ts]: Is this `TODO` still valid?

File: src/actions/viewProposal/validateActions.ts [hunk 5]
```diff
@@ -1,11 +1,25 @@
...
+  const actions: FetchActions<ValidateProposalResponse, any> = {
+    start: validateProposalStart,
+    success: validateProposalSuccess,
+    error: validateProposalError
+  };
+
+  return atkFetchCreator(buildRefreshRequest(proposalId), actions);
 };
 
+export const dismissValidationResult = () => dispatch => dispatch(dismissValidationResultCreator());
```
Comment (wojciech-kumon) [inline src/actions/viewProposal/validateActions.ts]: this wrapping is not needed
can be changed to  `export const dismissValidationResult = dismissValidationResultCreator;`
so it should be removed at all

File: src/components/viewProposal/ValidationResultModal.tsx [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+
+export default class ValidationResultModal extends React.PureComponent<ValidationResultModalProps> {
+  render() {
+    const {onClosed, validationData, formatMessage} = this.props;
+
+    const isModalOpen = validationData.isDisplayed;
+    const hasLoaded = !validationData.isValidating;
+
+    return (
+      <Modal isOpen={isModalOpen}>
```
Comment (wojciech-kumon) [inline src/components/viewProposal/ValidationResultModal.tsx]: adding `toggle={onClosed}` will allow closing modal with click outside of the modal

File: src/components/viewProposal/ValidationResultModal.tsx (hunk 1 above)
Comment (patrykbuzowiczsabre) [reply src/components/viewProposal/ValidationResultModal.tsx]: agree - simplified the conditions a bit

File: src/containers/ViewTripProposalPage.tsx (hunk 2 above)
Comment (patrykbuzowiczsabre) [reply src/containers/ViewTripProposalPage.tsx]: added some newlines and it looks way better now

File: src/containers/ViewTripProposalPage.tsx (hunk 3 above)
Comment (patrykbuzowiczsabre) [reply src/containers/ViewTripProposalPage.tsx]: eagle eye - thanks

File: src/__tests__/reducers/viewProposalReducer.test.ts (hunk 4 above)
Comment (patrykbuzowiczsabre) [reply src/__tests__/reducers/viewProposalReducer.test.ts]: it isn't - removing

File: src/actions/viewProposal/validateActions.ts (hunk 5 above)
Comment (patrykbuzowiczsabre) [reply src/actions/viewProposal/validateActions.ts]: right - too much of the thunks and callbacks. removing

File: src/components/viewProposal/ValidationResultModal.tsx (hunk 6 above)
Comment (patrykbuzowiczsabre) [reply src/components/viewProposal/ValidationResultModal.tsx]: worked like a charm, thanks

**Pull request merged**